tests/initializer_implicit_test.cc

```diff
@@ -0,0 +1,153 @@
+/**
+ * \file
+ */
+
+#include <gtest/gtest.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/smart_ptr/make_shared.hpp>
+
+#include "httpfake.h"
+#include "logging/logging.h"
+#include "primary/initializer.h"
+#include "primary/sotauptaneclient.h"
+#include "storage/fsstorage.h"
+#include "uptane/uptanerepository.h"
+#include "utilities/utils.h"
+
+/**
+ * \verify{\tst{185}} Verify that when using implicit provisioning, aktualizr
+ * halts if credentials are not available.
+ */
+TEST(UptaneImplicit, ImplicitFailure) {
+  Config config;
+  config.provision.device_id = "device_id";
+
+  TemporaryDirectory temp_dir;
+  config.storage.path = temp_dir.Path();
+  config.storage.uptane_metadata_path = "metadata";
+  config.storage.tls_cacert_path = "ca.pem";
+  config.storage.tls_clientcert_path = "client.pem";
+  config.storage.tls_pkey_path = "pkey.pem";
+  config.postUpdateValues();
+
+  auto storage = INvStorage::newStorage(config.storage);
+  HttpFake http(temp_dir.Path());
+  KeyManager keys(storage, config.keymanagerConfig());
+  std::map<std::string, std::shared_ptr<Uptane::SecondaryInterface> > secondaries;
+
+  Initializer initializer(config.provision, storage, http, keys, secondaries);
```

Comment:
Minor: every time we give an empty map to `Initializer`, we can just write `Initializer initializer(config.provision, storage, http, keys, {});` without having to define an empty secondaries map.

Reply:
Thanks!